src/client/pydaos/pydaos_core.py

```diff
@@ -107,12 +107,24 @@ def get(self, name):
     def __getitem__(self, name):
         return self.get(name)
 
-    def dict(self, name, v: dict = None):
+    def _get_object_id(self, cid):
+        """ Get the existing DAOS object class ID based on name.  """
+
+        # Default to OC_UNKNOWN (0), which will automatically select an object class.
+        if cid == "0":
+            return 0
+        if cid != "0":
+            return ObjClassID[cid].value
+
+    def dict(self, name, v: dict = None, cid="0"):
```

Comment:
This adds a pylint error
`invalid-name, Argument name "v" doesn't conform to snake_case naming style`
This one is easy to fix, but some others like `hi`, `lo`, etc. are more difficult. To appease the pylint GHA check, you can add this to the top of the file (under the module docstring)
```
# pylint: disable=invalid-name
```

Reply:
For new code please use valid names and only suppress warnings for methods/attributes that are part of the published API.

Reply:
dict() is an existing code, I just moved the function below.

Reply:
For what it's worth https://github.com/daos-stack/daos/pull/11825 fixes the pylint issues in this file, it might be an idea to cherry-pick the changes to this file in that PR into this PR.

Reply:
Done